src/shogun/lib/auto_initialiser.h

```diff
@@ -29,36 +34,93 @@ namespace shogun
 
 			~GammaFeatureNumberInit() override = default;
 
-			Any operator()() override
+			Any operator()() final
 			{
-				REQUIRE(m_kernel != nullptr, "m_kernel is not pointing to a CKernel object");
+				REQUIRE(
+				    m_kernel != nullptr,
+				    "m_kernel is not pointing to a CKernel object");
 
 				auto lhs = m_kernel->get_lhs();
 				switch (lhs->get_feature_class())
 				{
-					case EFeatureClass::C_DENSE:
-					case EFeatureClass::C_SPARSE:
-					{
-						auto dot_features = (CDotFeatures*)lhs;
-						return make_any(
-						    1.0 /
-						    (static_cast<float64_t>(
-								dot_features->get_dim_feature_space()) *
-								dot_features->get_std(false)[0]));
-					}
-					default:
-						return make_any(
-							1.0 /
-							static_cast<float64_t>(
-								((CDotFeatures*)lhs)->get_dim_feature_space()));
+				case EFeatureClass::C_DENSE:
+				{
+					auto dense_features = (CDenseFeatures<float64_t>*)lhs;
```

Comment:
I changed this to use the Eigen version of standard deviation because it should be more efficient?